pkg/runtime/misc/like.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package misc
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+type Liker interface {
+	Like(s string) bool
+}
+
+type liker struct {
+	pattern string
+}
+
+func NewLiker(pattern string) *liker {
+	return &liker{pattern: pattern}
+}
+
+func (l *liker) Like(s string) bool {
+	p := l.pattern
+	if strings.Index(p, "*") == -1 {
+		return like(p, s)
+	}
+
+	lc := "&"
```

Comment:
How about using an invisible character instead? For example, `\u0000` for `*` and '\u0001` for `?`.

Reply:
Think about the case of `LIKE '*&?abc_'`.